src/sgame/sg_votes.cpp

```diff
@@ -0,0 +1,1198 @@
+}
+
+/*
+==================
+G_Vote
+==================
+*/
+void G_Vote( gentity_t* ent, team_t team, bool voting )
+{
+	if ( !level.team[ team ].voteTime )
+	{
+		return;
+	}
+
+	if ( voting && ( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	if ( !voting && !( ent->client->pers.voted & ( 1 << team ) ) )
+	{
+		return;
+	}
+
+	ent->client->pers.voted |= 1 << team;
+
+	// TODO maybe refactor vote no/yes in one and only one variable and divide the NLOC by 2 ?
+
+	if ( voting )
+	{
+		level.team[ team ].voted++;
+	}
+	else
+	{
+		level.team[ team ].voted--;
+	}
+
+	if ( ent->client->pers.voteYes & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteYes++;
+		}
+		else
+		{
+			level.team[ team ].voteYes--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_YES + team, va( "%d", level.team[ team ].voteYes ) );
+	}
+
+	if ( ent->client->pers.voteNo & ( 1 << team ) )
+	{
+		if ( voting )
+		{
+			level.team[ team ].voteNo++;
+		}
+		else
+		{
+			level.team[ team ].voteNo--;
+		}
+
+		trap_SetConfigstring( CS_VOTE_NO + team, va( "%d", level.team[ team ].voteNo ) );
+	}
+}
+
+static constexpr char MARKER = '$';
+
+static bool IsEscapedMarker( Str::StringRef s, size_t pos )
+{
+	return pos + 1 < s.size() && s[ pos ] == MARKER && s[ pos + 1 ] == MARKER;
+}
+
+// Use a custom escape to avoid issues with Cmd::Escape unintentionally terminating quotes
+// for strings based on user input.
+static std::string VoteEscape( Str::StringRef s )
+{
+	if ( s.empty() )
+	{
+		return "";
+	}
+	std::string out;
+	out.reserve( s.size() * 2 );
+	for ( const auto& c : s )
+	{
+		out += '\\';
+		out += c;
+	}
+	return out;
+}
+
+// We only want to escape for commands that are executed. For display only, there is no injection
+// concern, so don't escape those.
+static std::string G_HandleVoteTemplate( Str::StringRef str, gentity_t* ent, team_t team,
+                                         std::string& cmd, std::string& arg, std::string& reason,
+                                         std::string& name, int clientNum, int id, bool escape )
+{
+	std::unordered_map<std::string, std::string> params = {
+		{"team",       BG_TeamNamePlural( team )  },
+		{ "arg",       arg						},
+		{ "reason",    reason                     },
+		{ "name",      name					   },
+		{ "slot",      std::to_string( clientNum )},
+		{ "namelogId", std::to_string( id )       },
+		{ "caller",    ent->client->pers.netname  },
+	};
+	std::string out;
+	out.reserve( str.size() + arg.size() + reason.size() );
+	size_t c = 0;
+	size_t s = 0;
+	size_t e = 0;
+	while ( c < str.size() )
+	{
+		s = str.find( MARKER, c );
+		if ( s != std::string::npos && !IsEscapedMarker( str, s ) )
+		{
+			e = str.find( MARKER, s + 1 );
+			if ( e != std::string::npos )
+			{
+				out += str.substr( c, s - c );
+				const auto& param = params[ str.substr( s + 1, e - s - 1 ) ];
+				out += Quote( Cmd::Escape( escape ? VoteEscape( param ) : param ) );
```

Comment:
Remove Cmd::Escape and Quote. The VoteEscape is to *replace* the other escaping functions which would allow injection attacks. And you don't want any escaping for display strings.